util.py

```diff
@@ -101,10 +101,14 @@ def short_hostname():
 
 
 class JSONEncoder(json.JSONEncoder):
+    _regexp_type = type(re.compile(''))
+
     def default(self, obj):
         if isinstance(obj, datetime.datetime):
             return {DATETIME_MAGIC_TOKEN: obj.strftime(FORMAT)}
-        return super(JSONEncoder, self).default(self, obj)
+        if isinstance(obj, self._regexp_type):
+            return None
```

Comment:
Using a placeholder like `return '<compiled re object>'` might make debugging easier in case someone tries to use the regexp after deserialization.